### PR TITLE
Fix OpRayQueryGenerateIntersectionKHR behavior

### PR DESCRIPTION
**Pull request by LLJJDD**
Problem:
In our GPURT implementation, rayQuery.RayTCurrent is relative to rayQuery.RayTMin, so for OpRayQueryGetIntersectionTKHR, we return (RayTCurrent + RayTMin) = Hit T relative to ray origin.

However, for OpRayQueryGenerateIntersectionKHR, we simply store the HitT value given by shader into rayQuery.RayTCurrent, and return it with RayTMin offset when OpRayQueryGetIntersectionTKHR is called again.

Fix:
For OpRayQueryGenerateIntersectionKHR, we store (HitT - rayQuery.RayTMin), so that rayQuery.RayTCurrent always represents offset to RayTMin.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 7b2597328daa62ec8eb3b401680258612ab04397
<details><summary>CTS tests (Failed: 0/208496)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 32525/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69503 (59.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 28359/69503 (40.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

File: llpc/lower/llpcSpirvLowerRayQuery.cpp [hunk 1]
```diff
@@ -864,7 +864,11 @@ template <> void SpirvLowerRayQuery::createRayQueryFunc<OpRayQueryGenerateInters
   storeAddr = m_builder->CreateGEP(
       rayQueryTy, rayQuery,
       {zero, m_builder->getInt32(RayQueryParams::Committed), m_builder->getInt32(RaySystemParams::RayTCurrent)});
-  m_builder->CreateStore(hitT, storeAddr);
+  Value *rayTMinAddr = m_builder->CreateGEP(rayQueryTy, rayQuery,
+                                            {m_builder->getInt32(0), m_builder->getInt32(RayQueryParams::RayTMin)});
```
Comment (nhaehnle) [inline llpc/lower/llpcSpirvLowerRayQuery.cpp:868]: You can use CreateConstGEP2_32 here.

**Review by nhaehnle — APPROVED** [review]
LGTM, but a nit/suggestion inline.

File: llpc/lower/llpcSpirvLowerRayQuery.cpp (hunk 1 above)
Comment (LLJJDD) [reply llpc/lower/llpcSpirvLowerRayQuery.cpp:868]: Yeah, and everywhere else that accesses ray query member. I will do that in another PR.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 8deee6997560afd2993334690678075555ee45be
<details><summary>CTS tests (Failed: 0/208496)</summary><ul><li>Built with version 1.3.5.2</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 36978/69503 (53.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 32525/69503 (46.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 41144/69503 (59.2%)</li>
<li>Failed: 0/69503 (0.0%)</li>
<li>Not Supported: 28359/69503 (40.8%)</li><li>Warnings: 0/69503 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 41178/69490 (59.3%)</li>
<li>Failed: 0/69490 (0.0%)</li>
<li>Not Supported: 28312/69490 (40.7%)</li><li>Warnings: 0/69490 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**